File: benchmark/perl/fact_approximation.pl
use Time::HiRes qw(time);
my $start = time * 1000;
my $res = 1.0;
for my $i (2..10000) { $res *= $i; }
my $end = time * 1000;
print "Result: $res
";
printf "Time: %.0fms
", $end - $start;

File: benchmark/src/lua/fact_approximation.lua
-- Note: Lua float calculations overflow to inf for factorial(10000)
local start = os.clock()
local result = 1.0
for i = 2, 10000 do result = result * i end
local finish = os.clock()
print("Result: " .. result)
print(string.format("Time: %.0fms", (finish - start) * 1000))